File: RB101_RB109_small_problems/easy_8/double_char_p2.rb
def consonant?(character)
  !character.count("a-z", "^aeiou").zero?
end

def double_consonants(input_string)
  doubled_characters = ""
  input_string.each_char do |char|
    case consonant?(char.downcase)
    when true 
      doubled_characters << (char * 2)
    when false
      doubled_characters << char
    end
  end
  p doubled_characters
end

p double_consonants('String') == "SSttrrinngg"
p double_consonants("Hello-World!") == "HHellllo-WWorrlldd!"
p double_consonants("July 4th") == "JJullyy 4tthh"
p double_consonants('') == ""

File: JS210_small_problems/easy_5/double_char_p2.js
function isConsonantCharacter(char) {
  const LETTER_REGEX = /[a-z]/ig;
  const VOWEL_REGEX = /[aeiou]/ig;

  return char.match(LETTER_REGEX) && !char.match(VOWEL_REGEX);
}

function doubleConsonants(str) {
  let doubledConsonantString = '';

  for (let char of str) {
    if (isConsonantCharacter(char)) {
      doubledConsonantString += char.repeat(2);
    } else {
      doubledConsonantString += char;
    }
  }

  return doubledConsonantString;
}

console.log(doubleConsonants('String'));          // "SSttrrinngg"
console.log(doubleConsonants('Hello-World!'));    // "HHellllo-WWorrlldd!"
console.log(doubleConsonants('July 4th'));        // "JJullyy 4tthh"
console.log(doubleConsonants(''));                // ""